spec/controllers/admin/event_schedules_controller_spec.rb

```diff
@@ -0,0 +1,143 @@
+                              room_id: nil,
+                              start_time: nil)
+
+          expect(JSON.parse(response.body)['status']).to eq("The event couldn't be scheduled. Event can't be blank. Room can't be blank. Start time can't be blank")
+        end
+      end
+    end
+
+    describe 'POST #update' do
+      context 'with valid attributes' do
+        it 'changes event schedule attributes' do
+          event = create(:event, program: conference.program)
+          room = create(:room, venue: venue)
+          patch :update, id: event_schedule.id, conference_id: conference.short_title, event_schedule:
+                 attributes_for(:event_schedule,
+                                schedule_id: schedule.id,
+                                event_id: event.id,
+                                room_id: room.id,
+                                start_time: conference.start_date)
+          event_schedule.reload
+          expect(event_schedule.schedule_id).to eq(schedule.id)
+          expect(event_schedule.event_id).to eq(event.id)
+          expect(event_schedule.room_id).to eq(room.id)
+          expect(event_schedule.start_time).to eq(conference.start_date)
+        end
+
+        it 'renders JSON without errors' do
+          patch :update, id: event_schedule.id, conference_id: conference.short_title, event_schedule:
+               attributes_for(:event_schedule,
+                              schedule_id: schedule.id,
+                              event_id: create(:event, program: conference.program).id,
+                              room_id: create(:room, venue: venue).id,
+                              start_time: conference.start_date)
+
+          expect(response).to be_success
+          expect(JSON.parse(response.body)['status']).to eq('ok')
+        end
+      end
+
+      context 'with invalid attributes' do
+        it 'does not save the event schedule to the database' do
+          expected = expect do
+            patch :update, id: event_schedule.id, conference_id: conference.short_title, event_schedule:
+                 attributes_for(:event_schedule,
+                                schedule_id: schedule.id,
+                                event_id: nil,
+                                room_id: nil,
+                                start_time: nil)
+          end
+          expected.to_not change { event_schedule }
+        end
+
+        it 'renders JSON with error' do
+          patch :update, id: event_schedule.id, conference_id: conference.short_title, event_schedule:
+               attributes_for(:event_schedule,
+                              schedule_id: schedule.id,
+                              event_id: nil,
+                              room_id: nil,
+                              start_time: nil)
+
+          expect(JSON.parse(response.body)['status']).to eq("The event couldn't be scheduled. Event can't be blank. Room can't be blank. Start time can't be blank")
```

Comment:
These are validation tests which we already cover in event_schedule_spec.rb
Just test the status code